### DISCUSSION TITLE
Processing webcal on sync or updating calendar data from cli

### DISCUSSION THREAD

**Discussion by leg7** [Kozea/Radicale#1563] [Q&A general]
Hello, my university provides an ics webcal synchronization link. The schedule however includes a lot of classes that don't concern me and it's hard to see what classes I have at a glance and scheduling anything quickly becomes a mess over overlapping rectangles.
To solve this I made a simple awk script that filters out events that match a regex you pass as a parameter. Then I would import the filtered .ics file to my calendar with a shell script. A cron job would automate this to sync every X hours.

I would like to know how I can process the university calendar after syncing in radicale. I tried filtering it and creating it in radicale from the command line but I could only create an empty calendar. Also I'm not sure my devices would still sync the calendar if I deleted it and recreated it everyday to filter it.

I looked into the plugin system documentation but I'm not sure if this could be done with it.

Does anyone experienced have any ideas how to solve this problem?

**Comment by pbiering** [comment]
Have you tried splitting the big ics into dedicated ones, filter out the ones you do not need and store the as-of-interest ics files into a folder of a collection. The change per day should be only resulting in add new interesting ics as new files or update adjusted ones. A regular cleanup can delete then really old events also.

Calender client can than sync that collection.

**Reply by leg7** [reply]
My schedule changes often as teachers can cancel classes or reschedule them last minute.
I would like the entire process to be automated so that I don't have to manually update the calendar everyday.

Do you think I could write a plugin that filters a webcalendar after every sync automatically?

**Reply by pbiering** [reply]
"last minute" sounds like a job has to check every minute for a change and then in case of changed rerun the defined work. Also the client need to pull in same cycle. I suggest first writing an external program doing the job as _radicale_ itself is not designed to run such jobs in general.

**Reply by leg7** [reply]
> "last minute" sounds like a job has to check every minute for a change and then in case of changed rerun the defined work. Also the client need to pull in same cycle. I suggest first writing an external program doing the job as _radicale_ itself is not designed to run such jobs in general.

Ok. Do you know how can I can update a radicale calendar through the command line?

**Reply by pbiering** [reply]
either one manage events using the WebDAV interface REPORT/PUT/DELETE or manipulate the files in the collection storage folder directly.
